explainability-core/src/main/java/org/kie/trustyai/explainability/model/Dataframe.java

```diff
@@ -0,0 +1,688 @@
+        }
+        this.data.set(column, values);
+    }
+
+    /**
+     * Set the {@link Value} at row and column.
+     *
+     * @param row Row to set the value.
+     * @param column Column to set the value.
+     * @param value The {@link Value} to set.
+     */
+    public void setValue(int row, int column, Value value) {
+        validateColumnIndex(column);
+        validateRowIndex(row);
+        data.get(column).set(row, value);
+    }
+
+    /**
+     * Get the {@link Value} at row and column.
+     *
+     * @param row Row to get the value.
+     * @param column Column to get the value.
+     * @return value The {@link Value}.
+     */
+    public Value getValue(int row, int column) {
+        validateColumnIndex(column);
+        validateRowIndex(row);
+        return data.get(column).get(row);
+    }
+
+    /**
+     * Drop a column from the dataframe.
+     * As when deleting an element for a {@link List}, the right-most column indices will be changed.
+     *
+     * @param column Index of the column to delete.
+     */
+    public void dropColumn(int column) {
+        validateColumnIndex(column);
+        data.remove(column);
+        metadata.remove(column);
+    }
+
+    /**
+     * Drop several columns from the dataframe
+     *
+     * @param columns Indices of the columns to drop
+     */
+    public void dropColumns(int... columns) {
+        Arrays.stream(columns).boxed().sorted(Comparator.reverseOrder()).forEach(this::dropColumn);
+    }
+
+    /**
+     * Drop several columns from the dataframe
+     *
+     * @param columns Indices of the columns to drop
+     */
+    public void dropColumns(List<Integer> columns) {
+        columns.stream().sorted(Comparator.reverseOrder()).forEach(this::dropColumn);
+    }
+
+    /**
+     * Return name of all columns
+     *
+     * @return A {@link List} with the column names
+     */
+    public List<String> getColumnNames() {
+        return metadata.names;
+    }
+
+    /**
+     * Return name of specified columns
+     *
+     * @return A {@link List} with the column names
+     */
+    public List<String> getColumnNames(List<Integer> indices) {
+        return indices.stream().map(metadata.names::get).collect(Collectors.toList());
+    }
+
+    /**
+     * Return the number of outputs.
+     *
+     * @return The number of outputs.
+     */
+    public int getOutputsCount() {
+        if (data.isEmpty()) {
+            return 0;
+        } else {
+            return (int) columnIndexStream().filter(i -> !metadata.inputs.get(i)).count();
+        }
+    }
+
+    /**
+     * Copy this dataframe as a new one.
+     *
+     * @return A {@link Dataframe}
+     */
+    public Dataframe copy() {
+        final Dataframe dataframe = new Dataframe();
+
+        dataframe.data = this.data.stream().map(ArrayList::new).collect(Collectors.toList());
+        dataframe.metadata = metadata.copy();
+        return dataframe;
```

Comment:
And use them here

Note: It is usually not safe to use the `List` returned by `.collect(Collectors.toList())`, the [Javadoc](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/stream/Collectors.html#toList()) specifies `There are no guarantees on the type, mutability, serializability, or thread-safety of the List returned;`
Please review other usage of `toList()` to verify if you expect users to modify the returned list

```suggestion
        return new Dataframe(
            this.data.stream().map(ArrayList::new).collect(Collectors.toList(ArrayList::new)),
            metadata.copy());
```